[Soh Enze] iP

sohenze opened the pull request:
DUKE
> Best task manager I've ever used - Me

Your friendly ~~neighbourhood~~ bot that helps you organise your tasks! 😄 

- Fast
- Simple
- Easy to use!

Start here:
1. Download it from [here](https://github.com/sohenze/ip)
2. Run it
3. Start adding your tasks!

Additional features:
- [x] "find" command to search for tasks
- [ ] Delete multiple tasks (soon)
- [ ] Reminders (soon)

Input your save file location here!
```java
public static void main(String[] args) {
    new Duke("data/TaskFile.txt").run();
}
```

Commit:
Add Gradle support

sohenze committed:
Level-1

sohenze committed:
Level-2

sohenze committed:
Level-3

sohenze committed:
Level-4

sohenze committed:
A-TextUiTesting

sohenze committed:
Level-5

sohenze committed:
Level-6

sohenze committed:
Add horizontal lines to exception messages

sohenze committed:
Clean up code and add invalid index exception

sohenze committed:
Fix code style

sohenze committed:
Level-7

sohenze committed:
Level-8

sohenze committed:
Merge branch 'branch-Level-7'

sohenze committed:
Merge branch 'branch-Level-8'

sohenze committed:
Stop tracking TaskFile.txt

sohenze committed:
Implement Storage class to load/write save file

sohenze committed:
Implement Ui, Storage, Parser, TaskList, Command classes

sohenze committed:
Merge branch 'add-gradle-support' of https://github.com/nus-cs2103-AY2223S1/ip

sohenze committed:
Add packages for exceptions, commands and tasks

sohenze committed:
Add tests for ToDo and Deadline classes

sohenze committed:
Fix ToDo empty desc exception bug

sohenze committed:
Fix time string bug in storage for deadline and event

sohenze committed:
Add javadocs

sohenze committed:
Fix coding standards

sohenze committed:
Fix event datetime bug

sohenze committed:
Implement Find command

sohenze committed:
Merge branch 'branch-Level-9'

sohenze committed:
Merge branch 'branch-A-CodingStandard'

sohenze committed:
Merge branch 'branch-A-JavaDoc'

# Conflicts:
#	src/main/java/duke/Storage.java
#	src/main/java/duke/task/Event.java
#	src/main/java/duke/task/Task.java
#	src/main/java/duke/task/TaskList.java

sohenze committed:
Add javadocs for find implementation

sohenze committed:
Change javadocs for Task constructor

sohenze committed:
Chang main class name in build.gradle

sohenze committed:
Add javafx dependencies to build.gradle

sohenze committed:
Update Duke exception messages

sohenze committed:
Create resources for Duke GUI

sohenze committed:
Update method signature for execute for commands

sohenze committed:
Delete ExitCommand

sohenze committed:
Update Parser after deleting ExitCommand

sohenze committed:
Update UI responses

sohenze committed:
Update Duke class for GUI

sohenze committed:
Create classes to handle GUI

sohenze committed:
Merge branch 'branch-Level-10'

sohenze committed:
Add checkstyle files

sohenze committed:
Change dukeResponse in Ui to private

sohenze committed:
Clean code according to checkstyle

sohenze committed:
Merge branch 'branch-A-CheckStyle'

Prearo commented on src/main/java/duke/task/Deadline.java:
```diff
@@ -0,0 +1,50 @@
+     * to be used when saving to a task file
+     */
+    @Override
+    public String toStringData() {
```
I don't like the term 'Data' here. Perhaps adding 'Storage' in the name may be clearer?
Same applies for Event, ToDo and Task.

Prearo commented on src/main/java/duke/task/Task.java:
```diff
@@ -0,0 +1,57 @@
+        return (this.isDone ? "X" : " ");
+    }
+
+    private String getStatusData() {
```
This could be integrated into toStringData().

Prearo commented on src/main/java/duke/Storage.java:
```diff
@@ -0,0 +1,109 @@
+ * tasks in a file on the hard drive
+ */
+public class Storage {
+    private static final int DESC_MARKER = 8;
```
I like the appropriate naming and smart use of offsets here.
However, consider adding a simple explanation for these magic numbers.

Prearo commented on src/main/java/duke/Storage.java:
```diff
@@ -0,0 +1,109 @@
+                String taskTime;
+
+                switch (taskType) {
+                case 'T':
```
I appreciate you following the standard outlined.
Personally though, some excess whitespace here can be done away with.

Prearo commented on src/main/java/duke/gui/MainWindow.java:
```diff
@@ -0,0 +1,55 @@
+
+import duke.Duke;
+import duke.Ui;
+import javafx.fxml.FXML;
```
It would be better to separate different categories of imports with a line break.

Prearo commented on src/main/java/duke/test/DeadlineTest.java:
```diff
@@ -0,0 +1,24 @@
+    }
+
+    @Test
+    void toStringData() {
```
Try to format your test methods as featureUnderTest_testScenario_expectedBehavior() as in the coding standard.

Prearo reviewed (commented):
Nicely done! The code is generally clean except some simple naming clarity issues.

sohenze committed:
Add image clip and label padding

sohenze committed:
Stop tracking images in git

sohenze committed:
Add comments for constants in Storage class

sohenze committed:
Change method names for Task classes

sohenze committed:
Add linebreak in imports

sohenze committed:
Change test class methods to follow naming format

sohenze committed:
Change main class name in build.gradle

sohenze committed:
Enable assertions in build.gradle

sohenze committed:
Reformat code in Storage class

sohenze committed:
Add asserts in Task, Deadline, Event classes

Constructors for Task, Deadline, Event classes do not check for
empty string parameters.

Constructors of Task, Event, Deadline classes should check for
empty string parameters to prevent an invalid Task from being
created at any given point.

Let's add assert statements to Task, Event, Deadline constructors
to ensure that no invalid Tasks are created.

No need for assert statements in ToDo class as it uses parent class
Task constructors exclusively without any additional parameters.

sohenze committed:
Create InvalidStorageDataException to represent exceptions while
parsing Storage data

Storage class does not use default clause to verify whether data
in storage file is valid.

A user that has directly edited the storage file might encounter
problems while running the program.

Let's check for invalid data while parsing the storage file.

sohenze committed:
Reformat nested code to improve readability

sohenze committed:
Add more checks for invalid storage file data

sohenze committed:
Add and refine comments in Storage class

sohenze committed:
Add line breaks to improve readability

sohenze committed:
Refine javadoc in Duke class

sohenze committed:
Refine javadocs in Main and Ui classes

sohenze committed:
Refactor code in TaskList for readability

sohenze committed:
Reformat code in Deadline and Event classes for readability

sohenze committed:
Refine code comments and line breaks for readability

sohenze committed:
Add line breaks for readability

sohenze committed:
Merge pull request #1 from sohenze/branch-A-Assertions

Branch-A-Assertions

sohenze committed:
Revert "Branch-A-Assertions"

sohenze committed:
Merge pull request #3 from sohenze/revert-1-branch-A-Assertions

Revert "Branch-A-Assertions"

sohenze committed:
Merge remote-tracking branch 'origin/master'

sohenze committed:
Add asserts in Task, Deadline, Event classes

Constructors for Task, Deadline, Event classes do not check for
empty string parameters.

Constructors of Task, Event, Deadline classes should check for
empty string parameters to prevent an invalid Task from being
created at any given point.

Let's add assert statements to Task, Event, Deadline constructors
to ensure that no invalid Tasks are created.

No need for assert statements in ToDo class as it uses parent class
Task constructors exclusively without any additional parameters.

sohenze committed:
Merge pull request #4 from sohenze/branch-A-Assertions

Add asserts in Task, Deadline, Event classes

sohenze committed:
Merge branch 'master' of https://github.com/sohenze/ip

sohenze committed:
Merge branch 'branch-A-CodeQuality'

# Conflicts:
#	src/main/java/duke/task/Deadline.java
#	src/main/java/duke/task/Event.java

sohenze committed:
Create gradle,yml

sohenze committed:
Rename gradle,yml to gradle.yml

sohenze committed:
Merge branch 'master' of https://github.com/sohenze/ip

sohenze committed:
Add EditDescCommand to update description of Tasks

Users have no way of editing existing Tasks.

Users may want to edit Tasks without deleting and creating a new
one.

Let's add an EditDescCommand to update the description of an
existing Task

Users may want to edit only a certain property of a Task

EditTimeCommand and EditStatusCommand will be added in the future

sohenze committed:
Merge pull request #5 from sohenze/branch-C-Update

Add EditDescCommand to update description of Tasks

sohenze committed:
Merge branch 'master' of https://github.com/sohenze/ip into branch-C-Update

sohenze committed:
Merge branch 'branch-C-Update'

sohenze committed:
Check for more invalid inputs

sohenze committed:
Print any errors when loading storage task file to user

mjgui commented on README.md:
```diff
@@ -13,7 +13,7 @@ Prerequisites: JDK 11, update Intellij to the most recent version.
 1. Configure the project to use **JDK 11** (not other versions) as explained in [here](https://www.jetbrains.com/help/idea/sdk.html#set-up-jdk).<br>
    In the same dialog, set the **Project language level** field to the `SDK default` option.
-3. After that, locate the `src/main/java/Duke.java` file, right-click it, and choose `Run Duke.main()` (if the code editor is showing compile errors, try restarting the IDE). If the setup is correct, you should see something like the below as the output:
+3. After that, locate the `src/main/java/duke.Duke.java` file, right-click it, and choose `Run duke.Duke.main()` (if the code editor is showing compile errors, try restarting the IDE). If the setup is correct, you should see something like the below as the output:
```
This was probably done by IntelliJ find and replace during refactoring, maybe consider changing it to `duke/Duke.java`?

mjgui commented on src/main/java/duke/Storage.java:
```diff
@@ -0,0 +1,125 @@
+ * tasks in a file on the hard drive
+ */
+public class Storage {
+    //Index at which description of task starts in storage
```
Perhaps consider adding a space after the `//` characters? i.e. `// Index ...`

mjgui commented on src/main/java/duke/command/EventCommand.java:
```diff
@@ -0,0 +1,42 @@
+     * Creates an Event Command object
+     */
+    public EventCommand(String desc, String time) throws InvalidDescriptionException, InvalidTimeException {
+        if (desc.isEmpty()) {
+            throw new InvalidDescriptionException();
+        }
+        if (time.isEmpty()) {
+            throw new InvalidTimeException();
+        }
```
I like the use of guard clauses here!

mjgui commented on src/main/java/duke/command/MarkCommand.java:
```diff
@@ -0,0 +1,35 @@
+     */
+    @Override
+    public void execute(TaskList taskList, Storage storage) {
+        taskList.markTask(this.isMark, this.index);
```
This isn't really a violation of coding standards, but perhaps consider omitting `this` when not necessary to follow the advanced coding guidelines?

mjgui commented on src/main/java/duke/test/DeadlineTest.java:
```diff
@@ -0,0 +1,23 @@
+import org.junit.jupiter.api.Test;
+
+import duke.task.Deadline;
+
+
+
```
Perhaps consider removing the excess whitespace here?

sohenze committed:
Change profile pictures for duke and user

sohenze committed:
Merge branch 'branch-A-CodeQuality'

# Conflicts:
#	src/main/java/duke/task/Deadline.java
#	src/main/java/duke/task/Event.java

sohenze committed:
Merge branch 'master' of https://github.com/sohenze/ip

sohenze committed:
Anchor send button to bottom right

sohenze committed:
Add application name to window title

sohenze committed:
Ignore docs/Ui.png

sohenze committed:
Update User Guide

sohenze committed:
Track docs/Ui.png

sohenze committed:
Update gui

sohenze committed:
Update Ui.png

sohenze committed:
Implement resizable Ui

sohenze committed:
Change image view circle clip

sohenze committed:
Adjust DialogBox spacing

sohenze committed:
Update header comments to fit coding standard
